### PR TITLE
feat(tableau-integrations): add support for tableau api

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes

- Add support for tableau api
- Tableau follows a different approach from what we currently support to generate tokens, which are later used to authenticate their REST API.
- To generate these tokens you will have to send a POST request with the following body 
```
{
    "credentials": {
        "personalAccessTokenName": "token_name",
        "personalAccessTokenSecret": "token_secret",
        "site": {
            "contentUrl": "content_url"
        }
    }
}
```
Also for proxy headers, you will have to append the following field for authentication
```
X-tableau-Auth: ${token}
```
## Issue ticket number and link
[EXT-100](https://linear.app/nango/issue/EXT-100/add-support-for-tableau)

### CODE REVIEW

File: docs-v2/integrations/all/tableau.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+-   [How to generate your Personal Access Token](https://help.tableau.com/current/pro/desktop/en-us/useracct.htm#create-and-revoke-personal-access-tokens)
+-   [Auth related docs](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_auth.htm#make-a-sign-in-request-with-a-personal-access-token)
+-   [API endpoints](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api.htm)
+
+<Tip>Need help getting started? Get help in the [community](https://nango.dev/slack).</Tip>
+
+## API gotchas
+
+- When creating a new Nango connection, please supply your Token name as your PAT NAME, Token secret as your PAT SECRET, and the content URL, which is the value that appears after `/site/` in the browser address bar for Tableau Cloud. This field is left blank if you are using Tableau Server.
```
Comment (bodinsamuel) [inline docs-v2/integrations/all/tableau.mdx]: Can you provide an example?

File: packages/shared/lib/services/proxy.service.ts [hunk 2]
```diff
@@ -416,6 +422,14 @@ class ProxyService {
                     };
                 }
                 break;
+            case 'TABLEAU':
+                {
+                    const token = config.token as TableauCredentials;
+                    headers = {
+                        'X-tableau-Auth': token
```
Comment (bodinsamuel) [inline packages/shared/lib/services/proxy.service.ts:429]: they are annoying :\

File: packages/frontend/lib/index.ts [hunk 3]
```diff
@@ -488,6 +522,12 @@ interface TBACredentials {
     oauth_client_secret_override?: string;
 }
 
+interface TableauCredentials {
+    pat_name: string;
+    pat_secret: string;
```
Comment (bodinsamuel) [inline packages/frontend/lib/index.ts:527]: There is no way to make that less specific to Tableau? feels like it's a regular PAT credentials with a dedicated header

**Review by bodinsamuel — COMMENTED** [review]
Looks good, I just wonder how generic can it be

File: docs-v2/integrations/all/tableau.mdx (hunk 1 above)
Comment (hassan254-prog) [reply docs-v2/integrations/all/tableau.mdx]: Updated this in the latest [commit](https://github.com/NangoHQ/nango/pull/2526/commits/9e893fa6ae4ee2cb36f97b5c892d8b76cb5444b4) 👍

File: packages/shared/lib/services/proxy.service.ts (hunk 2 above)
Comment (hassan254-prog) [reply packages/shared/lib/services/proxy.service.ts:429]: Yes very annoying 😅

File: packages/frontend/lib/index.ts (hunk 3 above)
Comment (hassan254-prog) [reply packages/frontend/lib/index.ts:527]: I was indeed considering that angle, where we might get another provider with a similar authentication method, so as to make it less specific to Tableau. However, it appears that this is a custom `auth_mode` specific to Tableau.

**Comment by khaliqgant** [comment]
@TBonnin can you review when you have a moment?

**Review by bodinsamuel — APPROVED** [review]

File: packages/shared/lib/utils/utils.ts [hunk 4]
```diff
@@ -94,6 +94,22 @@ export function parseTokenExpirationDate(expirationDate: any): Date {
...
+    const days = Number(daysStr);
+    const hours = Number(hoursStr);
+    const minutes = Number(minutesStr);
+
+    if (isNaN(days) || isNaN(hours) || isNaN(minutes)) {
+        return;
+    }
+
+    const millisecondsToAdd = (days * 24 * 60 * 60 + hours * 60 * 60 + minutes * 60) * 1000;
+    return new Date(Date.now() + millisecondsToAdd);
```
Comment (TBonnin) [inline packages/shared/lib/utils/utils.ts]: can be simplified slightly I think:
```
  const [days, hours, minutes] = timeStr.split(':').map(Number);
  
  if (days && hours && minutes) {
    const milliseconds = ((days * 24 + hours) * 60 + minutes) * 60 * 1000;
    return new Date(Date.now() + milliseconds);
  }
  
  return undefined;
```

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/utils/utils.ts (hunk 4 above)
Comment (hassan254-prog) [reply packages/shared/lib/utils/utils.ts]: Ah, yes, that simplifies this function, updated in this latest [commit](https://github.com/NangoHQ/nango/pull/2526/commits/364d3113a7a21b0770c381f3a2702b7729d1fd86). Thank you! 🙏

**Pull request merged**